perl/lib/Sanger/CGP/Star/Implement.pm

```diff
@@ -598,6 +599,23 @@ sub _which {
 	return $path;
 }
 
+sub _to_commandline_safe_tag_for_star {
+	# according to this: https://unix.stackexchange.com/a/398649
+	# preserving tag values is complicated
+	my $tag = shift;
+	# first skip back slashes
+	$tag =~ s/\\/\\\\/g;
+	# then skip other needed-to-skip characters.
+	# this will inevitably introduce an extra back slash into the tag if there's an '!', but for now no other way to skip '!' to prevent bash history expansion.
+	my @need_to_skips = ('$', '`', '"', '!');
+	foreach my $x (@need_to_skips) {
+		my $pattern = '['.$x.']';
+		$tag =~ s/$pattern/\\$x/g;
```

Comment:
You don't need to define the pattern, why not just:

```
$tag =~ s/[$x]/\\$x/g;
```

Also, you could just do:

```
$tag =~ s/([$`"!])/\\$1/g;
```